RFC/src/RFC-0310_AssetTemplates.md

```diff
@@ -0,0 +1,218 @@
+  friction / extra costs to contract execution (i.e. ethereum gas) to work around the halting problem.
+
+### Implementation
+
+Assets are created on the Tari network by issuing a `create_asset` instruction from a wallet or client and broadcasting
+it to the Tari Digital Assets Network (DAN).
+
+The instruction will contain the following information:
+(PKH = Public Key Hash)
+
+| Name                               | Type          | Description                                                                                                                   |
+|:-----------------------------------|:--------------|:------------------------------------------------------------------------------------------------------------------------------|
+| **Asset description**              |               |                                                                                                                               |
+| issuer                             | PKH           | The public key of the creator of the asset. See [issuer](#issuer)                                                             |
+| name                               | `string[64]`  | The name or identifier for the asset. See [Name and Description](#name-and-description)                                       |
+| description                        | `string[128]` | A short description of the asset - fits in a tweet. . See [Name and Description](#name-and-description)                       |
+| tldi                               | `string[16]`  | The Top Level Digital Issuer (TLDI) for the asset. The TLDI must be owned by the asset issuer, or empty / "Other" for no TLDI |
+| tldi_signature                     | `u256`        | Digital signature for the given TLDI                                                                                          |
+| template_number                    | `u64`         | The template number and version for this asset (e.g. 0x01 => Simple single-use token)                                         |
+| asset_expiry                       | `u64`         | A timestamp or block height after which the asset will automatically expire. Zero for arbitrarily long-lived assets           |
+| **Validation Committee selection** |               |                                                                                                                               |
+| committee_mode                     | `u8`          | The validation committee mode, either `PERMISSIONED` (0) or `PERMISSIONLESS` (1)                                              |
+| committee_parameters               | Object        |                                                                                                                               |
+| fee                                | `u64`         | The fee the issuer is paying, in nanoTari for the asset creation instruction                                                  |
```

Comment:
I guess it depends on whether this is the initial registration cost, or an ongoing pool of money to pay for the running of the asset.

The intent here was for the former, but we  may yet move to the latter model, in which case we would need to change the name

Reply:
My basis comment is that the variable name should be descriptive enough to capture the intent, thus to pre-fix or post-fix 'asset'. It could be a fee or a cost, depending on the model.

Reply:
I'll change it to `asset_creation_fee` to make it absolutely clear. If we change the model, we can amend